service/controller/clusterapi/v29/key/machine_deployment.go

```diff
@@ -1,29 +1,71 @@
 package key
 
 import (
+	"net"
 	"strconv"
 
 	g8sv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
 	"github.com/giantswarm/microerror"
 	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 
+	"github.com/giantswarm/aws-operator/pkg/annotation"
 	"github.com/giantswarm/aws-operator/pkg/label"
+	"github.com/giantswarm/aws-operator/service/network"
 )
 
 func WorkerClusterID(cr v1alpha1.MachineDeployment) string {
 	return cr.Labels[label.Cluster]
 }
 
+func WorkerSubnet(cr v1alpha1.MachineDeployment) string {
+	return cr.Annotations[annotation.MachineDeploymentSubnet]
+}
+
 // TODO this method has to be properly implemented and renamed eventually.
-func StatusAvailabilityZones(cr v1alpha1.MachineDeployment) []g8sv1alpha1.AWSConfigStatusAWSAvailabilityZone {
+func StatusAvailabilityZones(cr v1alpha1.MachineDeployment) ([]g8sv1alpha1.AWSConfigStatusAWSAvailabilityZone, error) {
```

Comment:
I don't follow - what is the todo about?
Why can we not sort this out now?

Reply:
We are limited to the current IPAM implementation. We maintain one cluster and basically one CF stack with two types, namely `Cluster` and `MachineDeployment`. The `ipam` resource allocates the cluster subnet. It does not yet allocated additional Node Pool subnets. So we only track the cluster subnet in the annotation right now as a temporary hack for transitioning reasons. Next step is to introduce the separate CF stack and sort out the reconciliation internally accordingly. Then, when the `ipam` resource of the `MachineDeployment` controller reconciles the Node Pool subnets, the implementation here is basically only to lookup the annotation and the function name should maybe be different.